src/components/ReserveForm.jsx

```diff
@@ -0,0 +1,100 @@
+import { useRef } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import '../assets/stylesheets/reservation.css';
+
+export default function ReserveForm({ trip, trips }) {
+  const navigate = useNavigate();
+  const user = JSON.parse(localStorage.getItem('user'));
+
+  const departureCity = useRef('');
+  const timeRef = useRef('');
+  const dateRef = useRef('');
+  const tripIdRef = useRef('');
+
+  const reserveHandler = () => {
+    if (tripIdRef.current.value === '' || departureCity.current.value === ''
+      || timeRef.current.value === '' || dateRef.current.value === '') {
+      // window.alert('Please fill all the fields');
+      return;
+    }
+
+    const token = localStorage.getItem('token');
+
+    fetch('https://capstone-backend-gz9j.onrender.com/api/v1/reservations', {
+      method: 'POST',
+      body: JSON.stringify({
+        user_id: user.id,
+        departure_city: departureCity.current.value,
+        date: dateRef.current.value,
+        time: timeRef.current.value,
+        trip_id: tripIdRef.current.value,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${token}`,
+      },
+    })
+      .then((res) => res.json())
+      .then((data) => data)
+      .catch(() => {
+        // window.alert(er.message);
```

Comment:
- [ ] Kindly, remove this all piece of commented code in order to improve your repo's cleanliness. 👍